roles/ceph-osd/tasks/start_osds.yml

```diff
@@ -32,5 +34,6 @@
   service:
     name: ceph-osd@{{ item }}
     state: started
+    enabled: true
```

Comment:
You are calling `ceph-disk prepare` only. That won't enable the service, it will just prepare the partitions. Once the partitions are prepared, udev detects them and takes over. It will call `ceph-disk trigger` and that step used to enable the service but that is what I am trying to change at the moment -- we use the same udev activation process on each boot as well and we definitely should not enable the service on each boot. Otherwise, we will disrespect the init system settings for enabled/disabled services.

I have a ceph PR for the ceph part of the change here:
https://github.com/ceph/ceph/pull/17209

In any case, I don't see how this line would hurt something even if it enabled the service ~twice? Maybe, I am missing something?

You can look at https://bugzilla.redhat.com/show_bug.cgi?id=1482201 for more details on why I am doing this.